cassdegrees/templates/student_edit.html

```diff
@@ -101,6 +104,44 @@ <h3>Courses</h3>
         </div>
     </div>
 
+    {# Popup for copying plan link #}
+    {% if render.popup %}
+    <div id="plan_popup" class="modal">
+        <div class="modal-background"></div>
+        <div class="modal-card card">
+            <header class="box-header">
+                Plan Link
+            </header>
+            <div class="box-solid box-has-footer">
+                <div class="anuform w-full">
+                    Your plan link is:
+                    <textarea id="plan_link" readonly class="w-full no-overflow" rows=10>{{ render.popup }}</textarea>
+                </div>
+                <div id="modal_content"></div>
+            </div>
+            <footer class="box-solid">
+                <input type="button" id="copy_to_clipboard" class="btn-uni-grad btn-small" value="Copy to Clipboard" />
+                <input type="button" id="close_modal" class="right btn-uni-grad btn-small" value="OK" />
+            </footer>
+        </div>
+        {# Script to allow interactivity in the menu #}
+        <script type="application/javascript">
+            var plan_link = document.getElementById("plan_link");
```

Comment:
Can this be extracted to a js script? In student_edit maybe?

Reply:
I can move this, but if I do it may confuse future readers as there is no indication that the three elements are using javascript. Should I still extract the code?

Reply:
Preferably, yeah. There are cross-site scripting issues down the line for inline Javascript - not relevant now, and there are a couple of other similar instances in the cost base, but something to keep in mind. Ideally the relevant elements would be commented.

Reply:
Okay, I added a fix to this in PR #243